kernel/src/main/java/org/kframework/kompile/Kompile.java

```diff
@@ -354,7 +354,7 @@ public static Module subsortKItem(Module module) {
     }
 
     public Rule parseAndCompileRule(CompiledDefinition compiledDef, String contents, Source source, Optional<Rule> parsedRule) {
-        Rule parsed = parsedRule.orElse(parseRule(compiledDef, contents, source));
+        Rule parsed = parsedRule.orElseGet(() -> parseRule(compiledDef, contents, source));
```

Comment:
What is this change for?

Reply:
orElse evaluates its argument unconditionally, which is pretty clearly not desired here. Without this change, parseRule will get called twice during k-compile-search-pattern even though one of those calls will be completely unused.

Reply:
:+1: